sdcm/utils/remote_logger.py

```diff
@@ -302,6 +313,131 @@ def _logger_cmd(self) -> str:
         return f"{cmd} >> {self._target_log_file} 2>&1"  # pylint: disable=protected-access
 
 
+class K8sClientLogger(LoggerBase):  # pylint: disable=too-many-instance-attributes
+    READ_REQUEST_TIMEOUT = 300
+    RECONNECT_DELAY = 30
+    CHUNK_SIZE = 1024
+
+    def __init__(self, node: "sdcm.cluster_k8s.BasePodContainer", target_log_file: str):
+        """
+        Reads logs from a k8s pod using k8s client API and forwards it to a file.
+
+        It is self-healing and will re-open the stream if it is closed and continue reading logs from place it stopped.
+        """
+        super().__init__(target_log_file)
+        logging.getLogger('parso.python.diff').setLevel(logging.WARNING)
+        self._node = node
+        self._parent_cluster = self._node.parent_cluster
+        self._last_log_time = ""
+        self._k8s_core_v1_api = KubernetesOps.core_v1_api(self._parent_cluster.k8s_cluster.get_api_client())
+        self._termination_event = ThreadEvent()
+        self._file_object = open(self._target_log_file, "w", encoding="utf-8")  # pylint: disable=consider-using-with
+        self._log_reader = None
+        self._thread = Thread(target=self._log_loop)
+
+    def start(self):
+        self._thread.start()
+
+    def stop(self, timeout=None):
+        self._log.info("Stopping logger for node %s", self._node.name)
+        self._termination_event.set()
+        self._thread.join(timeout)
+        self._file_object.close()
+
+    @retrying(n=20, sleep_time=3, allowed_exceptions=(ConnectionError, ))
+    def _open_stream(self) -> None:
+        self._log.debug("Opening log reader stream for node %s", self._node.name)
+        # read log since last read timestamp + 60 seconds to avoid missing logs due to some time difference between sct-runner and k8s
+        since_seconds = int(time.time() - isoparse(self._last_log_time).timestamp() +
```

Comment:
This approach looks very dangerous, because it depends on the time sync of SCT runner and target pod.
The diff may be more than 60 seconds and it is not guaranteed to have the same time zones.

I would really like to see usage of a single source of time -> either SCT runner or pod.

Reply:
fixed, used sct runner time